web-services/query/src/main/java/datawave/webservice/query/runner/RunningQuery.java

```diff
@@ -332,10 +332,13 @@ public ResultsPage next() throws Exception {
                 }
             }
         }
+        
         if (resultList.isEmpty()) {
             return new ResultsPage();
-        } else {
-            return new ResultsPage(resultList, ((hitPageByteTrigger || hitPageTimeTrigger) ? ResultsPage.Status.PARTIAL : ResultsPage.Status.COMPLETE));
+        } else if (iter.hasNext() && numResults < this.maxResults && currentPageCount < maxPageSize)
```

Comment:
The call to `iter.hasNext()` is causing a couple tests in `ExtendedRunningQueryTest` to fail. You'll need to update those tests to expect an additional call to this method. 

```
Error:  Failures: 
Error:    ExtendedRunningQueryTest.testNextMaxResults_HappyPathUsingDeprecatedConstructor:271 
  Unexpected method call TransformIterator.hasNext():
    TransformIterator.hasNext(): expected: 5, actual: 6
Error:    ExtendedRunningQueryTest.testNextWithDnResultLimit_HappyPathUsingDeprecatedConstructor:448 
  Unexpected method call TransformIterator.hasNext():
    TransformIterator.hasNext(): expected: 3, actual: 4
Error:    ExtendedRunningQueryTest.testNext_HappyPathUsingDeprecatedConstructor:184 
  Unexpected method call TransformIterator.hasNext():
    TransformIterator.hasNext(): expected: 3, actual: 4
Error:    ExtendedRunningQueryTest.testNext_NoResultsAfterCancellationUsingDeprecatedConstructor:322 
  Unexpected method call Query.getPagesize():
    Query.getUncaughtExceptionHandler(): expected: 3, actual: 1
    Query.getId(): expected: 2, actual: 1
    Query.getUserDN(): expected: 2, actual: 1
```

Reply:
Updated the tests please see new commit .